File: wasm/sample/src/xplpc/serializer/serializer.ts
import { DecodedRequest } from "../message/decoded-request";
import { Param } from "../message/param";

export interface Serializer {
    decodeRequest(data: string): DecodedRequest | undefined;
    encodeFunctionReturnValue(data: unknown): string;
    decodeFunctionReturnValue<T>(data: string): T | undefined;
    encodeRequest(functionName: string, ...params: Param[]): string;
}

File: swift/lib/Sources/Serializer/Serializer.swift
public protocol Serializer {
    func encodeRequest(_ functionName: String, _ params: [Param]) -> String
    func decodeFunctionReturnValue<T: Decodable>(_ data: String) -> T?
    func decodeRequest(_ data: String) -> DecodedRequest?
    func encodeFunctionReturnValue(_ data: ResponseData?) -> String
}
